### PR TITLE
Correct signature of allowRedirectBack to match code

### PR DESCRIPTION
**Pull request by oyeanuj**
The signature of `allowRedirectBack` in the docs doesn't match the code which tripped me up. Hope this saves someone else time :)

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ createOnEnter({
   ?authenticatingSelector: (state: Object, nextState: Object) => boolean,
   ?AuthenticatingComponent: ReactClass | ReactFunctionalComponent | string,
   ?wrapperDisplayName: string,
-  ?allowRedirectBack: string | (state: Object, nextState: Object) => boolean,
+  ?allowRedirectBack: string | (props: Object, redirectPath: String) => boolean,
```
Comment (mjrussell) [inline docs/API.md]: This should actually be `(nextState: Object, redirectPath: String)` - https://github.com/mjrussell/redux-auth-wrapper/blob/96414c997c27478d3be12b7ed133e99e65996338/src/history3/redirect.js#L51

Comment (oyeanuj) [reply docs/API.md]: Interesting. I was confused by this - https://github.com/mjrussell/redux-auth-wrapper/blob/aa1ea4bdde2dc5b25b9f12c247614444c485e5db/test/redirectBase-test.js#L214

**Comment by oyeanuj** [comment]
@mjrussell Made most of the changes, just one clarifying comment on `nextState` vs `props` for the signature.

What does `nextState` signify? This line seems to suggest that it is props being sent as the first parameter - https://github.com/mjrussell/redux-auth-wrapper/blob/aa1ea4bdde2dc5b25b9f12c247614444c485e5db/test/redirectBase-test.js#L214

**Comment by mjrussell** [comment]
`nextState` refers to how the React Router 3 documentation describes it - https://github.com/ReactTraining/react-router/blob/v3/docs/API.md#onenternextstate-replace-callback

Its not the `props` since there is no component here, its the react-router "state" which includes things like `location`, `params`, etc.

**Comment by oyeanuj** [comment]
@mjrussell Done, let me know if this is good to merge now.

**Comment by mjrussell** [comment]
LGTM!

**Pull request merged**